### ISSUE TITLE
推理使用的是causcal with prefix？

### ISSUE THREAD

**Issue by shyoulala** [THUDM/ChatGLM2-6B#64]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

我看了源码get_masks是causcal with prefix的矩阵，所以你们不是decoder only的标准样式吗？ 但是我用decoder only的标准样式训练反而效果比causcal with prefix方式好？？？？？请问你们训练的时候是用的decoder only的方式吗？还是causcal with prefix的方式

### Steps To Reproduce

![image](https://github.com/THUDM/ChatGLM2-6B/assets/23413415/647d760a-be90-4b3d-8ad4-4854aaebcbe2)

### Environment

```markdown
- OS:linux
- Python:
- Transformers:
- PyTorch:
- CUDA Support (`python -c "import torch; print(torch.cuda.is_available())"`) :
```

**Comment by duzx16** [comment]
在做对话训练的时候用的是 causal mask，你标的这段是给 kv cache 的 mask

**Comment by shyoulala** [comment]
我看这里full_attention_mask就是causcal with prefix的方式呀，求教，是我理解错误了吗？

**Comment by shyoulala** [comment]
> 

我看这里full_attention_mask就是causcal with prefix的方式呀，求教，是我理解错误了吗？

**Comment by Doufanfan** [comment]
> 我看这里full_attention_mask就是causcal with prefix的方式呀，求教，是我理解错误了吗？

kv cache是设置的微调prompt对应的相关数据，你不设置prompt我理解就是causal mask，这个时候训练是微调的模型参数；带prompt是prefix and causal，这个时候训练是微调的prompt部分的参数，模型参数是冻结的。所以感觉不设置prompt时候训练效果比设置prompt时候训练效果好也比较符合逻辑👁👁
